yt_dlp/utils/_utils.py

```diff
@@ -5121,16 +5121,21 @@ def clean_podcast_url(url):
     url = re.sub(r'''(?x)
         (?:
             (?:
-                chtbl\.com/track|
                 media\.blubrry\.com| # https://create.blubrry.com/resources/podcast-media-download-statistics/getting-started/
-                play\.podtrac\.com
+                play\.podtrac\.com|
+                chrt\.fm/track|
+                mgln.ai/e
             )/[^/]+|
+            (?:chtbl\.com/track)/[^\./]+|
+            (?:chtbl\.com/track)|
```

Comment:
```suggestion
            chtbl\.com/track(?:/[^/.]+)?|
```
&nbsp;
or, maybe it'd be better to move `chtbl\.com/track` back to where it was and do this:
```regex
            (?:
                chtbl\.com/track|
                media\.blubrry\.com| # https://create.blubrry.com/resources/podcast-media-download-statistics/getting-started/
                play\.podtrac\.com|
                chrt\.fm/track|
                mgln\.ai/e
            )(?:/[^/.]+)?|
```